compiler/src/dotty/tools/dotc/core/quoted/PickledQuotes.scala

```diff
@@ -33,7 +34,11 @@ object PickledQuotes {
   /** Transform the expression into its fully spliced Tree */
   def quotedExprToTree(expr: quoted.Expr[_])(implicit ctx: Context): Tree = expr match {
     case expr: TastyExpr[_] => unpickleExpr(expr)
-    case expr: LiftedExpr[_] => Literal(Constant(expr.value))
+    case expr: LiftedExpr[_] =>
+      expr.value match {
+        case value: Class[_] => ref(defn.Predef_classOf).appliedToType(classToType(value))
+        case value => Literal(Constant(value))
```

Comment:
Did you know that you can pass a `Type` to the constructor of `Constant` and if you do `Constant#tpe` on such a value you get an instance of `java.lang.Class` ?

Reply:
The miniphase `ClassOf` rewrites call to `classOf` into `Literal(Constant(...))`

Reply:
I will look at those.

Reply:
I can use the representation with `Literal(Constant(...))` from `ClassOf` though this representation is only expected after this phase. It seems better to create the `classOf[T]` and let `ClassOf` rewrite all of them at that phase.